src/run/run_environment/provider.rs

```diff
@@ -43,22 +44,15 @@ pub trait CIProvider {
     ///
     /// ```
     /// let provider = MyCIProvider::new();
-    /// assert_eq!(provider.get_provider_name(), "MyCIProvider");
+    /// assert_eq!(provider.get_run_environment_name(), "MyCIProvider");
     /// ```
-    fn get_provider_name(&self) -> &'static str;
+    fn get_run_environment_name(&self) -> &'static str;
```

Comment:
Since this is solely used for debug logging, remove this in favor of directly using the `get_run_environment`